app/controllers/ticket_transactions_controller.rb

```diff
@@ -0,0 +1,93 @@
+class TicketTransactionsController < ApplicationController
+  before_action :authenticate_user
+  before_action :set_ticketing, except: :index
+  protect_from_forgery except: [:hook]
+
+  def index
+    @ticket_transactions = TicketTransaction.pending(params[:id]).decorate
+  end
+
+  def set_ticketing
+    @ticketing ||= Ticketing.new
+  end
+
+  def checkout_ticket
+    transaction = TicketTransaction.find(params[:id])
+    @transaction = transaction.decorate
+    payers_email = @transaction.payers_email
+    amount = @ticketing.total_ticket_amount(@transaction)
+
+    if amount.to_i.eql?(0)
+      @ticketing.approve_transfer(transaction_id: transaction.id,
+                                  payer_email: payers_email)
+      redirect_to bookings_path
+    else
+      pay_info = { transaction: @transaction,
+                   amount: amount,
+                   return_path: ticket_transaction_hook_path }
+
+      redirect_to transaction.pay_pal_url(pay_info)
+    end
+  end
+
+  def hook
```

Comment:
This hook is already available in the booking controller as `paypal_hook`
It would be great to reuse instead of recreating a new one.